packages/tracking/src/user.spec.ts

```diff
@@ -0,0 +1,44 @@
+import fetch from 'jest-fetch-mock';
+import { fetchUser } from './user';
+
+const apiBaseUrl = 'https://www.codecademy.com';
+const authUser = {
+  "auth_token": "L2GFcDyBjcgtnDMo4q7n",
+  "experiments": {
+    "exp_le_get_unstuck": "variant",
+    "exp_weekly_streaks_for_new_users_2020_03_23": "control",
+    "exp_sorting_quiz_onboarding": "control"
+  },
+  "features": [
+    "catalog_home_page",
+    "catalog_home_new_and_noteworthy"
+  ],
+  "location": {
+    "geo_country": "US",
+    "in_eu": false
+  },
+  "id": "5eb1ae53b5915678de13fba2",
+  "email": "judah@codecademy.com",
+  "username": "css6822067325",
+  "name": null,
+  "profile_image_url": "https://www.gravatar.com/avatar/f3992b73c368230cf52ba5a100517ce3?s=140&d=retro",
+  "is_pro": true,
+  "is_premium": false
+};
+
+beforeAll(() => fetch.enableMocks());
+afterEach(() => fetch.resetMocks());
+afterAll(() => fetch.disableMocks());
+
+describe('fetchUser', () => {
+  test('retrieves a user', async () => {
+    fetch.once(JSON.stringify(authUser))
+    const user = await fetchUser(apiBaseUrl);
+
+    expect(user).toEqual(authUser);
+    expect(fetch.mock.calls.length).toEqual(1);
+    expect(fetch.mock.calls[0][0]).toBe(`${apiBaseUrl}/users/web`);
+    expect(fetch.mock.calls[0][1]!.method).toBe('GET');
+    expect(fetch.mock.calls[0][1]!.credentials).toBe('include');
```

Comment:
Nit: you're testing very precisely here, which is nice, but I don't think we need to worry too much about it making a lot of extra calls. What do you think about switching to something like this?

```suggestion
    expect(fetch).toHaveBeenLastCalledWith(`${apiBaseUrl}/users/web`, {
      credentials: 'include',
      method: 'GET',
    });
```

Reply:
There are some internal things like the headers, that I don't really want to check for.  In a way, I'm actually trying to make this a looser check.  Is there a way to test for a partial object (like `{ credentials, method }`?

Reply:
`toMatchObject`, maybe? https://jestjs.io/docs/en/expect